### ISSUE TITLE
Refine TPOT parameters

### ISSUE THREAD

**Issue by rhiever** [rhiever/tpot#252]
This is an extension to #206. Here, we will define specific parameter ranges for each individual operator's parameter, so we don't have to apply caps to the parameters any more. Many of these parameters are from auto-sklearn, with modifications where we know more about the parameter space.
# Classifiers
## AdaBoost

learning_rate: [1e-3, 1e-2, 1e-1, 0.5, 1.]
algorithm: ["SAMME.R", "SAMME"]
max_depth: range(1, 11)
## BernoulliNB

alpha: [1e-3, 1e-2, 1e-1, 1., 10., 100.]
fit_prior: [True, False]
## DecisionTree

criterion: ["gini", "entropy"]
max_depth: range(1, 11)
min_samples_split: range(2, 21)
min_samples_leaf: range(1, 21)
## ExtraTrees

criterion: ["gini", "entropy"]
max_features: np.arange(0, 1.01, 0.05)
min_samples_split: range(2, 21)
min_samples_leaf: range(1, 21)
bootstrap: ["True", "False"]
## GradientBoosting

learning_rate: [1e-3, 1e-2, 1e-1, 0.5, 1.]
max_depth: range(1, 11)
min_samples_split: range(2, 21)
min_samples_leaf: range(1, 21)
subsample: np.arange(0.05, 1.01, 0.05)
max_features: np.arange(0, 1.01, 0.05)
## KNeighbors

n_neighbors: range(1, 101)
weights: ["uniform", "distance"]
p: [1, 2]
## LinearSVC

penalty: ["l1", "l2"]
loss: ["hinge", "squared_hinge"]
dual: [True, False]
tol: [1e-5, 1e-4, 1e-3, 1e-2, 1e-1]
C: [1e-4, 1e-3, 1e-2, 1e-1, 0.5, 1., 5., 10., 15., 20., 25.]

In preprocess, disallow:
- penalty = 'l1' and loss = 'hinge' (set loss = 'squared_hinge')
- dual = False and penalty = 'l2' and loss = 'hinge' (set loss = 'squared_hinge')
- dual = False and penalty = 'l1' (set dual = True)
## LogisticRegression

penalty: ["l1", "l2"]
C: [1e-4, 1e-3, 1e-2, 1e-1, 0.5, 1., 5., 10., 15., 20., 25.]
dual: [True, False]

In preprocess, disallow:
- dual = False and penalty = 'l1' (set dual = True)
## MultinomialNB

alpha: [1e-3, 1e-2, 1e-1, 1., 10., 100.]
fit_prior: [True, False]
## PassiveAggressive

Drop this classifier
## RandomForest

criterion: ["gini", "entropy"]
max_features: np.arange(0, 1.01, 0.05)
min_samples_split: range(2, 21)
min_samples_leaf: range(1, 21)
bootstrap: ["True", "False"]
## XGBClassifier

max_depth: range(1, 11)
learning_rate: [1e-3, 1e-2, 1e-1, 0.5, 1.]
subsample: np.arange(0.05, 1.01, 0.05)
min_child_weight: range(1, 21)
# Regressors
## AdaBoost

learning_rate: [1e-3, 1e-2, 1e-1, 0.5, 1.]
loss: ["linear", "square", "exponential"]
max_depth: range(1, 11)
## DecisionTree

max_depth: range(1, 11)
min_samples_split: range(2, 21)
min_samples_leaf: range(1, 21)
## ElasticNetCV

l1_ratio: np.arange(0.0, 1.01, 0.05)
tol: [1e-5, 1e-4, 1e-3, 1e-2, 1e-1]
## ExtraTrees

max_features: np.arange(0, 1.01, 0.05)
min_samples_split: range(2, 21)
min_samples_leaf: range(1, 21)
bootstrap: ["True", "False"]
## GradientBoosting

loss: ["ls", "lad", "huber", "quantile"]
learning_rate: [1e-3, 1e-2, 1e-1, 0.5, 1.]
max_depth: range(1, 11)
min_samples_split: range(2, 21)
min_samples_leaf: range(1, 21)
subsample: np.arange(0.05, 1.01, 0.05)
max_features: np.arange(0, 1.01, 0.05)
alpha: [0.75, 0.8, 0.85, 0.9, 0.95, 0.99]
## KNeighbors

n_neighbors: range(1, 101)
weights: ["uniform", "distance"]
p: [1, 2]
## LassoLarsCV

No parameters
## LinearSVR

loss: ["epsilon_insensitive", "squared_epsilon_insensitive"]
dual: [True, False]
tol: [1e-5, 1e-4, 1e-3, 1e-2, 1e-1]
C: [1e-4, 1e-3, 1e-2, 1e-1, 0.5, 1., 5., 10., 15., 20., 25.]
epsilon: [1e-4, 1e-3, 1e-2, 1e-1, 1.]

In preprocess, disallow:
- dual = False and loss = "epsilon_insensitive" (set dual = True)
## PassiveAggressive

Drop this regressor
## RandomForest

max_features: np.arange(0, 1.01, 0.05)
min_samples_split: range(2, 21)
min_samples_leaf: range(1, 21)
bootstrap: ["True", "False"]
## RidgeCV

No parameters
## XGBRegressor

max_depth: range(1, 11)
learning_rate: [1e-3, 1e-2, 1e-1, 0.5, 1.]
subsample: np.arange(0.05, 1.01, 0.05)
min_child_weight: range(1, 21)
